### PR TITLE
Feature/82 tab bar

### PR DESCRIPTION
**Pull request by cjbatin**
## Ticket name and link

#82 

## Preview

With feature flag on

No Changes with feature flag off

## Description

Adds a tab bar with a feature flag (turned on) for development of additional screens

## Things to check

Checklist:

- [ ] Two reviews
- [ ] Manually tested iOS
- [ ] Manually tested Android
- [ ] (Optional) Updated the [Confluence docs](https://anddigitaltransformation.atlassian.net/wiki/spaces/ML/pages/4398940396/MCDBA+Mobile+App)

### CODE REVIEW

File: src/navigation/HomeContainer/HomeContainer.tsx [hunk 1]
```diff
@@ -72,17 +79,40 @@ function HomeContainer(): React.JSX.Element {
...
-      <Stack.Screen
-        name="Bookings"
-        component={BookingScreen}
-        options={{
-          headerRight: logoutButton,
-        }}
-      />
-    </Stack.Navigator>
-  );
+  console.log(featureFlags.tabBarEnabled);
```
Comment (Jake-Sturgeon) [inline src/navigation/HomeContainer/HomeContainer.tsx:82]: Remove console . log

File: src/navigation/HomeContainer/HomeContainer.tsx [hunk 2]
```diff
@@ -72,17 +79,40 @@ function HomeContainer(): React.JSX.Element {
...
+  if (featureFlags.tabBarEnabled === true) {
```
Comment (SSmithSky) [inline src/navigation/HomeContainer/HomeContainer.tsx:83]: Shouldn't need the === true. 

This should be:
```
if (featureFlags.tabBarEnabled) {
```

File: src/navigation/HomeContainer/HomeContainer.tsx [hunk 3]
```diff
@@ -72,17 +79,40 @@ function HomeContainer(): React.JSX.Element {
...
+    console.log('========');
+    console.log('TAB BAR ENABLED');
```
Comment (SSmithSky) [inline src/navigation/HomeContainer/HomeContainer.tsx:85]: First time reviewing the Wharf spaces app so I'm not sure if this is standard but are these console.logs needed? It seems like either they've accidentally been left in or they would clog up the console. Seems a little superfluous 😄

Comment (cjbatin) [reply src/navigation/HomeContainer/HomeContainer.tsx:85]: Yeah I need to remove these. I have done in my current WIP that I need to commit. They were in when I was struggling to get the tests to work

File: src/navigation/HomeContainer/HomeContainer.tsx (hunk 2 above)
Comment (cjbatin) [reply src/navigation/HomeContainer/HomeContainer.tsx:83]: Ahh does the work? I thought if you did that it just checked if it existed. Typescript isn't my first language so happy to remove the === true

Comment (SeanSmithAND) [reply src/navigation/HomeContainer/HomeContainer.tsx:83]: Sorry was accidentally logged in to my Sky account.

Comment (SeanSmithAND) [reply src/navigation/HomeContainer/HomeContainer.tsx:83]: Yes it should work. 

As you're probably aware if() is basically checking truthy values. In javascript this means...
```
0 // falsey
1 // truthy
(anything above 1) // truthy
null // falsey
undefined // falsey
```

From the above you can hopefully see what would pass in the if statement. 

Now with the `!` this inverts the value so `0` becomes `true` and the potential `undefined` (which could be returned due to your optional chaining is also `true`. 

Hope this makes sense?

File: src/types/FeatureFlags.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export type FeatureFlags = {
+  tabBarEnabled: Boolean;
```
Comment (edwinsimpson) [inline src/types/FeatureFlags.ts:2]: use primitive `boolean` for the feature flag - the TS type system doesn't force objects to their primitive types, and it's recommended to use primitives

File: src/navigation/HomeContainer/HomeContainer.tsx (hunk 2 above)
Comment (cjbatin) [reply src/navigation/HomeContainer/HomeContainer.tsx:83]: Thanks, makes sense and done!

File: src/firebase/remoteConfig/fetchRemoteConfig.ts [hunk 5]
```diff
@@ -10,6 +11,14 @@ export function getIsDemoLoginEnabled(): boolean {
   return remoteConfig().getBoolean('isDemoLoginEnabled');
 }
 
+export function getFeatureFlags(): FeatureFlags | undefined {
+  const featureFlags = remoteConfig().getString('featureFlags');
+  if (featureFlags) {
+    return JSON.parse(featureFlags);
+  }
+  return undefined;
```
Comment (Jake-Sturgeon) [inline src/firebase/remoteConfig/fetchRemoteConfig.ts:19]: Rather than undefined can we return a default value

Comment (cjbatin) [reply src/firebase/remoteConfig/fetchRemoteConfig.ts:19]: Not in this case as the default value is the one we have defined in the app elsewhere. Don't want to have multiple sources of truth

File: src/navigation/HomeContainer/HomeContainer.tsx [hunk 6]
```diff
@@ -70,19 +76,38 @@ function HomeContainer(): React.JSX.Element {
...
-        component={BookingScreen}
-        options={{
-          headerRight: logoutButton,
-        }}
-      />
-    </Stack.Navigator>
-  );
+  if (featureFlags?.tabBarEnabled) {
+    return (
+      <Tab.Navigator
```
Comment (Jake-Sturgeon) [inline src/navigation/HomeContainer/HomeContainer.tsx:84]: What happens if we navigate within a tab?

Comment (cjbatin) [reply src/navigation/HomeContainer/HomeContainer.tsx:84]: We don't at the moment so there is no way for me to test anything here. I suspect tweaks will be needed once we do further development on this. I have tested the presentation sheets we use and alerts and they all work okay though

**Comment by cjbatin** [comment]
Closing as recreated in another PR

**Pull request closed without merging**